jsonschema/validators.py

```diff
@@ -862,6 +868,44 @@ def resolve_remote(self, uri):
             self.store[uri] = result
         return result
 
+    def store_subschema(self, schema, last_schema=None, last_url=None):
+        """
+        Using $id or id with $ref, save subschema to self.store
+
+        Arguments:
+
+            schema:
+
+                The referring schema.
+
+            last_schema:
+
+                The referring upper level schema.
+
+            last_url:
+
+                Save the last URL.
+        """
+        if not isinstance(schema, dict) \
```

Comment:
Keep  ```"id" : "http://json-schema.org/draft-03/schema#", "id" : "http://json-schema.org/draft-04/schema#", "id" : "http://json-schema.org/draft-06/schema#", "id" : "http://json-schema.org/draft-07/schema#"```  unchanged in self.store

Reply:
@Julian This if statement block is replaced with the following statement, there will be no problem with python3 and above， but python2 will report  error `exceptions.AttributeError: NoneType' object has no attribute 'find'`
```
if not isinstance(schema, dict) or schema.get(u"$schema") in meta_schemas:
     return
```
In order to maintain compatibility with python2, I keep the current if statement block